### PR TITLE
Plans & Placements Updates

### PR DESCRIPTION
**Pull request by eraderna**
Updating of plans & placements to address issues

- updates from entries added to #":update-.*" columns of issues dataset CSV file

### CODE REVIEW

File: src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj [hunk 1]
```diff
@@ -585,3 +602,136 @@
...
+
+
+
+;;; # CSV file read
+(defn csv-file->ds
+  "Read columns required to construct census
+   from CSV file of `plans-placements-on-census-dates` into a dataset."
+  [filepath]
+  (tc/dataset filepath
+              {:column-whitelist (map name key-columns-for-census)
```
Comment (otfrom) [inline src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj:615]: allowlist

File: src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj [hunk 2]
```diff
@@ -585,3 +602,136 @@
...
+                                         :age-at-start-of-school-year [:int8 parse-long]
+                                         :ncy-nominal                 [:int8 parse-long]})}))
+
+(defn updates-csv-file->ds
+  "Read columns required to update columns for census
+   from CSV file of `plans-placements-on-census-dates-issues`
+   with update columns filled in into a dataset."
+  [filepath]
+  (tc/dataset filepath
+              {:column-whitelist (map name [:person-table-id
```
Comment (otfrom) [inline src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj:630]: allowlist

File: notebooks/sen2_blade_csv_eda.clj [hunk 3]
```diff
@@ -115,3 +119,16 @@
...
+^{::clerk/visibility {:code :hide, :result :hide}}
+(comment ;; clerk build
+  (let [in-path            (str "notebooks/" (clojure.string/replace (str *ns*) #"\.|-" {"." "/" "-" "_"}) ".clj")
+        out-path           (str wk-dir (clojure.string/replace (str *ns*) #"^.*\." "") ".html")]
+    (clerk/build! {:paths    [in-path]
+                   :ssr      true
+                   :bundle   true
+                   :out-path "."})
+    (.renameTo (io/file "./index.html") (io/file out-path)))
+  )
```
Comment (otfrom) [inline notebooks/sen2_blade_csv_eda.clj:134]: I want to use the adroddiad stuff for this, but I don't want to bring in adroddiad as a dependency

File: notebooks/sen2_blade_csv_plans_placements.clj [hunk 4]
```diff
@@ -137,13 +135,29 @@
...
+(comment
+  (let [ds              plans-placements-on-census-dates
+        file-name-stem  (tc/dataset-name ds)
+        col-name->label sen2-blade-csv-plans-placements/plans-placements-on-census-dates-col-name->label]
+    (tc/write! (tc/dataset {:column-number (iterate inc 1)
+                            :column-name   (map name   (tc/column-names ds))
+                            :column-label  (map col-name->label (tc/column-names ds))})
+               (str wk-dir file-name-stem "-col-labels.csv"))
+    (tc/write! ds
+               (str wk-dir file-name-stem ".csv")))
```
Comment (otfrom) [inline notebooks/sen2_blade_csv_plans_placements.clj:151]: This feels like the important output and the handover to the next step, IIUC.

File: notebooks/sen2_blade_csv_plans_placements.clj [hunk 5]
```diff
@@ -36,21 +37,21 @@
 
 ;;; ## Parameters
 ;;; ### Working directory
-;; For output files:
-^{::clerk/viewer clerk/md}
+^{::clerk/visibility {:code :show, :result :hide}
```
Comment (otfrom) [inline notebooks/sen2_blade_csv_plans_placements.clj:40]: worth adding a section above saying what is taken in (blade export files?) and what is produced at the end? (basic csv extract, issues file, updated csv extract after modification file applied?)

I think I get a bit confused as the notebook sort of needs to be run in two bits. Once to do the extract, then you look at it, and then the second part to do the updates to the raw extract. Do I have that correctly?

**Review by otfrom — CHANGES_REQUESTED** [review]
Just a few small changes (a bit of doc in the template and moving to allowlist now that tmd supports it).

File: src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj (hunk 1 above)
Comment (eraderna) [reply src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj]: TMD deps bumped to 7.010 (TC still at 7.007) and anachronistic keywords updated.

File: src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj (hunk 2 above)
Comment (eraderna) [reply src/witan/sen2/return/person_level/blade_export/csv/plans_placements.clj]: done

File: notebooks/sen2_blade_csv_eda.clj (hunk 3 above)
Comment (eraderna) [reply notebooks/sen2_blade_csv_eda.clj:134]: I also didn't want to introduce adroddiad dependencies, so duplicated some of the required age/ncy stuff in a local ns `witan.sen2.ncy` (introduced in PR#4).

**Review by otfrom — APPROVED** [review]
thx!

File: notebooks/sen2_blade_csv_plans_placements.clj (hunk 4 above)
Comment (eraderna) [reply notebooks/sen2_blade_csv_plans_placements.clj:162]: UUC: I've added some explanation and code to read it back in as part of  the update step.

File: notebooks/sen2_blade_csv_plans_placements.clj (hunk 5 above)
Comment (eraderna) [reply notebooks/sen2_blade_csv_plans_placements.clj:51]: I've added an explanatory section at the top of the notebook and described how in practice the "update" step would be done elsewhere.

**Review by otfrom — APPROVED** [review]

**Pull request merged**